apps/src/util/HttpClient.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  AUTHENTICITY_TOKEN_HEADER,
+  getAuthenticityToken,
+} from './AuthenticityTokenStore';
+
+/**
+ * Get a JSON response from the given endpoint and
+ * return it as the specified type. Can also perform
+ * response validation if provided a validator function.
+ */
+export async function getJson<ResponseType>(
+  endpoint: string,
+  init?: RequestInit,
+  validator?: ResponseValidator<ResponseType>
+): Promise<GetResponse<ResponseType>> {
+  const response = await fetch(endpoint, init);
+  if (!response.ok) {
+    throw new Error(response.status + ' ' + response.statusText);
+  }
+
+  const json = await response.json();
+  let value = json;
```

Comment:
for my information: how is `value` inferred to be `ResponseType` here (assuming it is because it could be returned as-is)?

Reply:
I think because `json` is `any`, `value` gets narrowed to `ResponseType` in the return statement? That wasn't super clear to me too though, I'll add an explicit type for clarity.